### DISCUSSION TITLE
grblHAL tool change with other senders?

### DISCUSSION THREAD

**Discussion by that-jim** [grblHAL/core#243] [Q&A] [answered]
Is it possible to use the tool change functions in grblHAL with other senders? I'm using a OB BlackBoxX32 and the Openbuilds Control sender. I have a test file running using mode 3 and it pauses for tool changes but does not probe the bit length. Any help is appreciated. I did test this with io sender and it works, I'm just more accustom to the OpenBuilds Control and would like to keep using it. Any help is appreciated.

**Comment by terjeio** [comment, marked as answer]
> Is it possible to use the tool change functions in grblHAL with other senders?

Yes, provided that they implement the tool [change protocol](https://github.com/grblHAL/core/wiki/Manual-tool-change-protocol). However, I am not aware of any that does. You will have to ask the sender developers to add support?

**Reply by that-jim** [reply]
Thanks, I will ask the sender developer.

**Reply by muhammadelmogy** [reply]
bCNC Support Tool Change!

**Reply by terjeio** [reply]
> bCNC Support Tool Change!

AFAIKT not the grblHAL tool change protocol.

**Reply by muhammadelmogy** [reply]
I am not really sure about it, I have used it several times in the past for tool changing.
According to the main page, it is a grblHAL sender!
